code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm

```diff
@@ -770,11 +770,14 @@
 	maxcharge = 1000000000
 
 /obj/machinery/power/secured_capacitor/attack_hand(mob/user as mob)
+	if(user.lying || istype(user,/mob/dead/observer))
```

Comment:
isn't this what the `ghost_read` var is for?

Reply:
unless I'm mistaken just setting the var to 0 is enough to prevent attack_hand from being called at all, if it's a ghost doing it

Reply:
`ghost_read` is checked for in the parent's `attack_hand()`, so I wouldn't expect it to prevent the call, but I'll test it anyway.